code/modules/reagents/metabolism.dm

```diff
@@ -207,19 +207,20 @@
 			addiction_list.Remove(R)
 			continue
 
-		addiction_list[R] += 1
-		if(!parent.chem_effects[CE_PURGER])
-
+		if(!parent.chem_effects[CE_PURGER] && ishuman(parent))
+			addiction_list[R] += 1
 			switch(addiction_list[R])
-				if(1 to 10)
+				if(1 to 20)
 					R.addiction_act_stage1(parent)
-				if(10 to 20)
-					R.addiction_act_stage2(parent)
 				if(20 to 30)
-					R.addiction_act_stage3(parent)
+					R.addiction_act_stage2(parent)
 				if(30 to 40)
+					R.addiction_act_stage3(parent)
+				if(40 to 50)
+					R.addiction_act_stage4(parent)
+				if(50 to INFINITY && ((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim))))
 					R.addiction_act_stage4(parent)
-				if(40 to INFINITY)
+				if(50 to INFINITY && !((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim))))
```

Comment:
```suggestion
				else
```

Reply:
ELSE DOES NOT WORK i tried unfortunately
that else checks for every IF above, so, for example, if you are at stage 1, you aren't at stage 2(duh), which makes the 'else' proc

Reply:
Bruh. Okay, try this then.
```suggestion
				if(50 to INFINITY)
					if((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim)))
						R.addiction_act_stage4(parent)
					else
						R.addiction_end(parent)
						addiction_list.Remove(R)
```

Reply:
yeh should wok